velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -47,11 +47,9 @@ RowTypePtr getNonPartitionTypes(
   const auto& dataSize = dataCols.size();
   childNames.reserve(dataSize);
   childTypes.reserve(dataSize);
-  for (uint32_t i = 0; i < inputType->size(); i++) {
-    if (std::find(dataCols.cbegin(), dataCols.cend(), i) != dataCols.cend()) {
-      childNames.push_back(inputType->nameOf(i));
-      childTypes.push_back(inputType->childAt(i));
-    }
+  for (int dataCol : dataCols) {
```

Comment:
I'm actually okay either implementation. Both make its own sense.

I see a strong assumption made is dataCols holds indices of input columns in the original order. (Elements of dataCols are in ascending order.) And the assumption comes from HiveInsertTableHandle. Can you document this assumption somewhere, for the dataChannels_ member variable or here for the func? It took me a while to see this assumption.